src/util/PixelBuffer.cxx

```diff
@@ -137,10 +137,12 @@ PixelBuffer & PixelBuffer::colorMagicPixels(RGBAColor color)
             g = pixel >> 16,
             b = pixel >> 8,
             a = pixel;
-    if(r != b) continue;
+    if(r != b) {
+      continue;
+    }
     RGBAColor magic {r, g, b, a};
-    float s_n = overlay(magic.saturation(), s_t);
-    float l_n = overlay(magic.lightness(), l_t);
+    float s_n = overlay(s_t, magic.saturation());
```

Comment:
This is another fix to magic pixels. The overlay was being done in the wrong order.